## Introduction
This notebook aims to extract tickers of companies working in the **Biotechnology** industry. In order to do so, a dataset
containing information about all the companies in the US Stock Market was used. This dataset divides companies into 69 industries, including Biotechnology.

[View Initial Dataset on Kaggle] (https://www.kaggle.com/datasets/marketahead/all-us-stocks-tickers-company-info-logos)

## Import Packages & Data

In [26]:
import pandas as pd

In [27]:
external_dataset = pd.read_csv('data/companies.csv')
external_dataset

ticker                     company name                   short name  \
0         A        Agilent Technologies Inc.                      Agilent   
1        AA                Alcoa Corporation                        Alcoa   
2      AABA                      Altaba Inc.                       Altaba   
3       AAC                AAC Holdings Inc.                          AAC   
4      AADR  AdvisorShares Dorsey Wright ADR  AdvisorShares Dorsey Wright   
...     ...                              ...                          ...   
6363    ZTS              Zoetis Inc. Class A                       Zoetis   
6364   ZUMZ                      Zumiez Inc.                       Zumiez   
6365    ZUO               Zuora Inc. Class A                        Zuora   
6366   ZYME                   Zymeworks Inc.                    Zymeworks   
6367   ZYNE     Zynerba Pharmaceuticals Inc.      Zynerba Pharmaceuticals   

                            industry  \
0     Medical Diagnostics & Research   
1                    Metals & Mining   
2                   Asset Management   
3              Health Care Providers   
4                                NaN   
...                              ...   
6363              Drug Manufacturers   
6364    Retail - Apparel & Specialty   
6365            Application Software   
6366                   Biotechnology   
6367                   Biotechnology   

                                            description  \
0     Agilent Technologies Inc is engaged in life sc...   
1     Alcoa Corp is an integrated aluminum company. ...   
2     Altaba Inc is an independent, non-diversified,...   
3     AAC Holdings Inc provides inpatient and outpat...   
4     The investment seeks long-term capital appreci...   
...                                                 ...   
6363  Zoetis Inc is a developer and manufacturer of ...   
6364  Zumiez Inc is a multi-channel specialty retail...   
6365  Zuora Inc provides cloud-based software on a s...   
6366  Zymeworks Inc is a clinical-stage biopharmaceu...   
6367  Zynerba Pharmaceuticals Inc together with its ...   

                                      website      logo  \
0                      http://www.agilent.com     A.png   
1                        http://www.alcoa.com    AA.png   
2                       http://www.altaba.com  AABA.png   
3     http://www.americanaddictioncenters.org       NaN   
4                http://www.advisorshares.com  AADR.png   
...                                       ...       ...   
6363                    http://www.zoetis.com   ZTS.png   
6364                    http://www.zumiez.com  ZUMZ.png   
6365                    https://www.zuora.com   ZUO.png   
6366                 http://www.zymeworks.com  ZYME.png   
6367                   http://www.zynerba.com  ZYNE.png   

                         ceo                 exchange    market cap  \
0        Michael R. McMullen  New York Stock Exchange  2.421807e+10   
1     Roy Christopher Harvey  New York Stock Exchange  5.374967e+09   
2        Thomas J. Mcinerney     Nasdaq Global Select  4.122368e+10   
3      Michael T. Cartwright  New York Stock Exchange  6.372010e+07   
4                        NaN                NYSE Arca  1.031612e+08   
...                      ...                      ...           ...   
6363        Juan Ramon Alaix  New York Stock Exchange  4.205627e+10   
6364       Richard M. Brooks     Nasdaq Global Select  6.150368e+08   
6365               Tien Tzuo  New York Stock Exchange  2.304595e+09   
6366             Ali Tehrani  New York Stock Exchange  5.042878e+08   
6367           Armando Anido     NASDAQ Global Market  9.994437e+07   

                  sector               tag 1  \
0             Healthcare          Healthcare   
1        Basic Materials     Basic Materials   
2     Financial Services  Financial Services   
3             Healthcare          Healthcare   
4                    NaN                 NaN   
...                  ...                

## Dataset Overview

In [28]:
external_dataset.columns

Index(['ticker', 'company name', 'short name', 'industry', 'description',
       'website', 'logo', 'ceo', 'exchange', 'market cap', 'sector', 'tag 1',
       'tag 2', 'tag 3'],
      dtype='object')

In [29]:
external_dataset.industry.value_counts()

Asset Management        598
Biotechnology           491
Banks                   449
Application Software    305
REITs                   226
                       ... 
Personal Services        12
Health Care Plans        10
Tobacco Products          8
Medical Distribution      7
Truck Manufacturing       6
Name: industry, Length: 69, dtype: int64

As can be seen, there are **491** companies working in the Biotechnology industry

## Extract Data

In [30]:
# biotech -> companies whose industry feature equals Biotechnology
biotech = external_dataset.loc[external_dataset.industry == 'Biotechnology']

In [31]:
biotech

ticker                   company name               short name  \
24     ABEO       Abeona Therapeutics Inc.      Abeona Therapeutics   
28     ABIO            ARCA biopharma Inc.           ARCA biopharma   
34     ABUS  Arbutus Biopharma Corporation        Arbutus Biopharma   
37     ACAD    ACADIA Pharmaceuticals Inc.   ACADIA Pharmaceuticals   
42     ACER         Acer Therapeutics Inc.        Acer Therapeutics   
...     ...                            ...                      ...   
6348   ZIOP          ZIOPHARM Oncology Inc        ZIOPHARM Oncology   
6352   ZLAB                Zai Lab Limited          Zai Lab Limited   
6359   ZSAN      Zosano Pharma Corporation            Zosano Pharma   
6366   ZYME                 Zymeworks Inc.                Zymeworks   
6367   ZYNE   Zynerba Pharmaceuticals Inc.  Zynerba Pharmaceuticals   

           industry                                        description  \
24    Biotechnology  Abeona Therapeutics Inc is a clinical-stage bi...   
28    Biotechnology  ARCA biopharma Inc is a biopharmaceutical comp...   
34    Biotechnology  Arbutus Biopharma Corp is a biopharmaceutical ...   
37    Biotechnology  ACADIA Pharmaceuticals Inc is a biotechnology ...   
42    Biotechnology  Acer Therapeutics Inc is a pharmaceutical comp...   
...             ...                                                ...   
6348  Biotechnology  ZIOPHARM Oncology Inc is a biotechnology compa...   
6352  Biotechnology  Zai Lab Ltd is a biopharmaceutical company. Th...   
6359  Biotechnology  Zosano Pharma Corp is a clinical-stage special...   
6366  Biotechnology  Zymeworks Inc is a clinical-stage biopharmaceu...   
6367  Biotechnology  Zynerba Pharmaceuticals Inc together with its ...   

                                website      logo                         ceo  \
24    http://www.abeonatherapeutics.com  ABEO.png                Joao Siffert   
28         http://www.arcabiopharma.com       NaN          Michael R. Bristow   
34            http://www.arbutusbio.com       NaN                 Mark Murray   
37          http://www.acadia-pharm.com  ACAD.png            Stephen R. Davis   
42               https://www.acertx.com  ACER.png             Chris Schelling   
...                                 ...       ...                         ...   
6348            http://www.ziopharm.com  ZIOP.png  Laurence James Neil Cooper   
6352       http://www.zailaboratory.com       NaN                     Ying Du   
6359        http://www.zosanopharma.com       NaN              John P. Walker   
6366           http://www.zymeworks.com  ZYME.png                 Ali Tehrani   
6367             http://www.zynerba.com  ZYNE.png               Armando Anido   

                     exchange    market cap      sector       tag 1  \
24      NASDAQ Capital Market  3.643781e+08  Healthcare  Healthcare   
28      NASDAQ Capital Market  5.848104e+06  Healthcare  Healthcare   
34       Nasdaq Global Select  2.207994e+08  Healthcare  Healthcare   
37       Nasdaq Global Select  2.839192e+09  Healthcare  Healthcare   
42      NASDAQ Capital Market  2.387585e+08  Healthcare  Healthcare   
...                       ...           ...         ...         ...   
6348    NASDAQ Capital Market  3.583108e+08  Healthcare  Healthcare   
6352     NASDAQ Global Market  1.410893e+09  Healthcare  Healthcare   
6359    NASDAQ Capital Market  2.744220e+07  Healthcare  Healthcare   
6366  New York Stock Exchange  5.042878e+08  Healthcare  Healthcare   
6367     NASDAQ Global Market  9.994437e+07  Healthcare  Healthcare   

              tag 2 tag 3  
24    Biotechnology   NaN  
28    Biotechnology   NaN  
34    Biotechnology   NaN  
37    Biotechnology   NaN  
42    Biotechnology   NaN  
...             ...   ...  
6348  Biotechnology   NaN  
6352  Biotechnology   NaN  
6359  Biotechnology   NaN  
6366  Biotechnology   NaN  
6367  Biotechnology   NaN  

[491 rows x 14 columns]

In [32]:
# ticker and company name features are the only attributes need!
cols = ['ticker', 'company name']
biotech = biotech[cols]
biotech

ticker                   company name
24     ABEO       Abeona Therapeutics Inc.
28     ABIO            ARCA biopharma Inc.
34     ABUS  Arbutus Biopharma Corporation
37     ACAD    ACADIA Pharmaceuticals Inc.
42     ACER         Acer Therapeutics Inc.
...     ...                            ...
6348   ZIOP          ZIOPHARM Oncology Inc
6352   ZLAB                Zai Lab Limited
6359   ZSAN      Zosano Pharma Corporation
6366   ZYME                 Zymeworks Inc.
6367   ZYNE   Zynerba Pharmaceuticals Inc.

[491 rows x 2 columns]

In [37]:
# Reset index to start from 0 
biotech = biotech.reset_index().drop(columns = 'index')
biotech

ticker                   company name
0     ABEO       Abeona Therapeutics Inc.
1     ABIO            ARCA biopharma Inc.
2     ABUS  Arbutus Biopharma Corporation
3     ACAD    ACADIA Pharmaceuticals Inc.
4     ACER         Acer Therapeutics Inc.
..     ...                            ...
486   ZIOP          ZIOPHARM Oncology Inc
487   ZLAB                Zai Lab Limited
488   ZSAN      Zosano Pharma Corporation
489   ZYME                 Zymeworks Inc.
490   ZYNE   Zynerba Pharmaceuticals Inc.

[491 rows x 2 columns]

In [38]:
# Export biotech data frame as a .csv file
biotech.to_csv('data/biotech_companies.csv', index=False)